server/test/sc/server/plugins/TestGame.kt

```diff
@@ -0,0 +1,96 @@
+package sc.server.plugins
+
+import org.slf4j.LoggerFactory
+import sc.api.plugins.IGameState
+import sc.api.plugins.exceptions.TooManyPlayersException
+import sc.framework.plugins.ActionTimeout
+import sc.framework.plugins.Player
+import sc.framework.plugins.RoundBasedGameInstance
+import sc.protocol.responses.ProtocolMessage
+import sc.server.helpers.TestTeam
+import sc.shared.*
+import java.util.*
+
+class TestGame : RoundBasedGameInstance<TestPlayer?>(TestPlugin.TEST_PLUGIN_UUID) {
+    private val state = TestGameState()
+    override fun onRoundBasedAction(fromPlayer: Player, data: ProtocolMessage) {
+        /*
+         * NOTE: Checking if right player sent move was already done by
+         * {@link sc.framework.plugins.RoundBasedGameInstance#onAction(Player, Object)}.
+         * There is no need to do it here again.
+         */
+
+        if (data is TestMove) {
+            data.perform(state)
+            next(if (state.currentPlayer === TestTeam.RED) state.red else state.blue)
+        }
+    }
+
+    override fun checkWinCondition(): WinCondition? {
+        return if (this.round > 1) {
+            logger.info("Someone won")
+            WinCondition(if (state.state % 2 == 0) TestTeam.RED else TestTeam.BLUE, TestWinReason.WIN)
+        } else null
+    }
+
+    override fun onPlayerJoined(): Player {
+        if (players.size < 2) {
+            return if (players.isEmpty()) {
+                state.red = TestPlayer(TestTeam.RED)
+                players.add(state.red)
+                state.red
+            } else {
+                state.blue = TestPlayer(TestTeam.BLUE)
+                players.add(state.blue)
+                state.blue
```

Comment:
this can be simplified (DRY) a lot by creating a temporary variable for the player

Reply:
but state.red and state.blue are already player variables